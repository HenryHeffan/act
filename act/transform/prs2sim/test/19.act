/*
   x.e connected to y.e within a subprocess through different ports

x unique, done

x.<unique port>, done

x is the canonical name, but x is also connected to a, b, c, d.

*/


defchan c1 <: chan(bool) (bool x) { }

defproc sub (c1 a, b)
{
  a.x = b.x;

  prs {
    [keeper=0] a.x -> b.x-
  }
}

defproc sub2 (c1 a)
{
  bool b;
  prs {
    [keeper=0] a.x -> b-
  }
}

defproc foo ()
{
   c1 l, m;

   sub s(l,m);

   sub2 t(l);
   sub2 u(m);
}

foo f;
