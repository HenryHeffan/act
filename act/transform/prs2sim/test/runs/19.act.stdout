| units: 3 tech: dummytech format: MIT
n f/u/a/x GND f/u/b 2 5
n f/t/a/x GND f/t/b 2 5
n f/s/a/x GND f/s/a/x 2 5
= f/m/x f/u/a/x
= f/m/x f/s/b/x
= f/s/a/x f/s/b/x
= f/l/x f/t/a/x
= f/l/x f/s/a/x
= f/l/x f/m/x
= Vdd Vdd!
= GND GND!
